### PR TITLE
feat(cdk): dev-mode command

### PR DESCRIPTION
**Pull request by afiune**
## Summary

Wanna develop a new CDK component? Run the command:
```
lacework component dev <component_name>
```

Signed-off-by: Salim Afiune Maya <afiune@lacework.net>

## How did you test this change?

https://user-images.githubusercontent.com/5712253/196309255-37f17404-620f-4ae3-982e-acc7b81986c8.mp4

## Issue

https://lacework.atlassian.net/browse/ALLY-1186

### CODE REVIEW

**Review by hazedav — APPROVED** [review]

File: cli/cmd/component.go [hunk 1]
```diff
@@ -404,3 +419,74 @@ func runComponentsDelete(_ *cobra.Command, args []string) (err error) {
...
+	}
+
+	component, found := cli.LwComponents.GetComponent(args[0])
+	if !found {
+		component = &lwcomponent.Component{
+			Name: args[0],
+		}
+
+		if component.UnderDevelopment() {
+			return errors.New("component already under development.")
```
Comment (dmurray-lacework) [inline cli/cmd/component.go:439]: nit: do you want to return an error here or just nil and output info?

File: cli/cmd/component.go [hunk 2]
```diff
@@ -76,6 +78,15 @@ var (
...
 		RunE:    runComponentsDelete,
 	}
+
+	// componentsDevModeCmd represents the dev sub-command inside the components command
+	componentsDevModeCmd = &cobra.Command{
+		Use:    "dev <component>",
+		Hidden: true,
+		Short:  "Enter development mode of a new or existing component",
+		Args:   cobra.ExactArgs(1),
+		RunE:   runComponentsDevMode,
```
Comment (dmurray-lacework) [inline cli/cmd/component.go:88]: nit: I know this is hidden, but a 'long' description with more details on cmd usage would be nice.

**Review by dmurray-lacework — COMMENTED** [review]
Question on this, if I have an existing component and I enter dev mode on it, how do I then exit dev mode?

File: cli/cmd/component.go [hunk 3]
```diff
@@ -404,3 +419,74 @@ func runComponentsDelete(_ *cobra.Command, args []string) (err error) {
...
+	cli.LwComponents, err = lwcomponent.LoadState(cli.LwApi)
+	cli.StopProgress()
+	if err != nil {
+		return errors.Wrap(err, "unable to load components")
+	}
+
+	component, found := cli.LwComponents.GetComponent(args[0])
+	if !found {
+		component = &lwcomponent.Component{
+			Name: args[0],
```
Comment (dmurray-lacework) [inline cli/cmd/component.go:435]: Should there be any restrictions on a component name. For example I could pass in "!"

**Pull request merged**